String bytes doc updates

arezaii opened the pull request:
This PR updates the documentation for string and bytes as 
described in https://github.com/Cray/chapel-private/issues/2422.

TESTING:

After `make cleanall` can:

- [x] `make all`
- [x] `make docs`
- [x] `make mason`
- [x] full paratest pass

Reviewed by @bmcdonald3, thanks!

Signed-off-by: Ahmad Rezaii ahmad.rezaii@hpe.com

bmcdonald3 commented on modules/internal/Bytes.chpl:
```diff
@@ -136,7 +136,8 @@ module Bytes {
   */
-  inline proc createBytesWithBorrowedBuffer(x: c_string, length=x.size) {
+  inline proc createBytesWithBorrowedBuffer(x: c_string,
+                                            length=x.size) : bytes {
```
Do these line breaks cause any changes to the docs or was it just > 80 characters?

arezaii replied:
They don't affect the docs, they were just > 80 chars after adding the return type. You can view the docs if you download and extract them from the artifact here: https://github.com/chapel-lang/chapel/pull/18502/checks?check_run_id=3761565776. (Artifacts link is in the upper right as you view that page)

bmcdonald3 reviewed (approved):
Looks good. I just have a few questions:
1. Have we decided that the Chapel-team endorsed style to specify return types is by putting a space before and after the `:`? (i.e., `proc foo() : int`)
2. The line breaks seem a bit random and not totally consistent with where the break happens; how did you decide when to put a line break in parameters here? (Mainly since I never use them and am curious when I'm supposed to)

arezaii commented:
> 1. Have we decided that the Chapel-team endorsed style to specify return types is by putting a space before and after the `:`? (i.e., `proc foo() : int`)

I don't know that there is a style guide for this. I just added the space before to visually balance it. The space after is needed so that the docs generate a space between the method signature and return type...but maybe that's not the desired style either? 

> 2. The line breaks seem a bit random and not totally consistent with where the break happens; how did you decide when to put a line break in parameters here? (Mainly since I never use them and am curious when I'm supposed to)

I added the line breaks when a method that I edited ended up > 80 chars. In that case, I wrapped the arguments and indented to  the column where the first argument appears. They _should_ be lined up on that column, but if you find one that isn't, please point it out for fixin'.

bmcdonald3 commented:
Thanks for clarifying, I didn't see any that didn't looked lined up, I just wasn't sure they all were at the same length, but your comment reassures me. I think it's good to go.
